ftw/upgrade/command/__init__.py

```diff
@@ -142,7 +144,25 @@ def __call__(self):
         args = self.parser.parse_args()
         configure_logging(args)
         setattr(args, 'parser', self.parser)
-        args.func(args)
+        if getattr(args, 'all_sites', False):
+            info = {}
+            while True:
+                try:
+                    with capture() as out:
+                        args.func(args)
+                    output = out.getvalue().strip()
+                    if getattr(args, 'json', False):
+                        # "[]" must be turned into [].
+                        output = json.loads(output)
+                    info[args.picked_site] = output
+                except StopIteration:
+                    break
+            if getattr(args, 'json', False):
+                print(json.dumps(info))
+            else:
+                print(info)
```

Comment:
Yes, extra commits may help while reviewing, but squashing is nice in the end.
I think it depends on the changes; I sometimes squash before merging and sometimes do interactive rebasing for small changes.
Since it depends on the changes, I leave the decision to the pull-requestor :stuck_out_tongue_winking_eye: